pilot/pkg/bootstrap/server.go

```diff
@@ -797,7 +801,21 @@ func (s *Server) initRegistryEventHandlers() {
 	s.ServiceController().AppendServiceHandler(serviceHandler)
 
 	if s.configController != nil {
-		configHandler := func(_, curr config.Config, event model.Event) {
+		configHandler := func(old config.Config, curr config.Config, event model.Event) {
+			if old.Generation == curr.Generation && curr.Generation != 0 {
+				// generation 0 indicates this is an ephemeral config generated by istio
+				if curr.GroupVersionKind == gvk.WorkloadEntry {
+					oldCond := modelstatus.GetConditionFromSpec(old, modelstatus.ConditionHealthy)
```

Comment:
This seems to be leaking details about a specific status field in workload entry, when we are in a codepath dealing with generic configs, which seems not ideal to me

Reply:
What would you propose?  I agree that this is not ideal, but the problem is that all configs behave the same way with regard to status pushes, except WE... I don't see a way to handle this generically...